src/main/java/Duke.java

```diff
@@ -1,10 +1,282 @@
+
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private static final ArrayList<String> Data = new ArrayList<>();
+    private static boolean hasData = false;
+    public static void main(String[] args) throws FileNotFoundException {
```

Comment:
Avoid long methods (>30 lines) if possible. You can extract some of the logic into methods for better readability.